vic/drivers/cesm/src/cesm_put_data.c

```diff
@@ -91,14 +91,84 @@ vic_cesm_put_data()
         l2x_vic[i].l2x_Fall_lwup = 0;
         l2x_vic[i].l2x_Fall_evap = 0;
         l2x_vic[i].l2x_Fall_swnet = 0;
-        // l2x_vic[i].l2x_Fall_fco2_lnd = 0;
-        // l2x_vic[i].l2x_Fall_flxdst1 = 0;
-        // l2x_vic[i].l2x_Fall_flxdst2 = 0;
-        // l2x_vic[i].l2x_Fall_flxdst3 = 0;
-        // l2x_vic[i].l2x_Fall_flxdst4 = 0;
-        // l2x_vic[i].l2x_Fall_flxvoc = 0;
+        l2x_vic[i].l2x_Fall_fco2_lnd = 0;
+        l2x_vic[i].l2x_Fall_flxdst1 = 0;
+        l2x_vic[i].l2x_Fall_flxdst2 = 0;
+        l2x_vic[i].l2x_Fall_flxdst3 = 0;
+        l2x_vic[i].l2x_Fall_flxdst4 = 0;
+        l2x_vic[i].l2x_Fall_flxvoc = 0;
         l2x_vic[i].l2x_Flrl_rofliq = 0;
-        // l2x_vic[i].l2x_Flrl_rofice = 0;
+        l2x_vic[i].l2x_Flrl_rofice = 0;
+
+        // populate reference values
+
+        // 10m wind, VIC: m/s, CESM: m/s
+        l2x_vic[i].l2x_Sl_u10 = out_data[i][OUT_WIND][0];
+
+        // 2m reference temperature, VIC: C, CESM: K
+        l2x_vic[i].l2x_Sl_tref = out_data[i][OUT_AIR_TEMP][0] + CONST_TKFRZ;
+
+        // 2m reference specific humidity, VIC: kg/kg, CESM: g/g
+        l2x_vic[i].l2x_Sl_qref = CONST_EPS *
+                                 out_data[i][OUT_VP][0] /
+                                 out_data[i][OUT_PRESSURE][0];
+
+        // band-specific quantities
+        // note that these include a veg correction (AreaFactor)
+        // that is already in the put_data routine
+
+        // temperature, VIC: K, CESM: K
+        l2x_vic[i].l2x_Sl_t = out_data[i][OUT_RAD_TEMP][0];
+
+        // albedo, VIC: fraction, CESM: fraction
+        // Note: VIC does not partition its albedo, thus all types are
+        // the same value
+        // TBD: this will be fixed in a subsequent PR
+        albedo = out_data[i][OUT_ALBEDO][0];
+
+        // albedo: direct, visible
+        l2x_vic[i].l2x_Sl_avsdr = albedo;
+
+        // albedo: direct , near-ir
+        l2x_vic[i].l2x_Sl_anidr = albedo;
+
+        // albedo: diffuse, visible
+        l2x_vic[i].l2x_Sl_avsdf = albedo;
+
+        // albedo: diffuse, near-ir
+        l2x_vic[i].l2x_Sl_anidf = albedo;
+
+        // snow height, VIC: mm, CESM: m
+        // convert to VIC units
+        l2x_vic[i].l2x_Sl_snowh = out_data[i][OUT_SWE][0] / MM_PER_M;
+
+        // net shortwave, VIC: W/m2, CESM: W/m2
+        l2x_vic[i].l2x_Fall_swnet = out_data[i][OUT_SWNET][0];
+
+        // longwave up, VIC: W/m2, CESM: W/m2
+        // adjust sign for CESM sign convention
+        l2x_vic[i].l2x_Fall_lwup = -1 *
+                                   (out_data[i][OUT_LWDOWN][0] -
+                                    out_data[i][OUT_LWNET][0]);
```

Comment:
Nope - we have `LWDOWN`, `LWNET` and the band-specific version of `LWNET` in `out_data`.